### PR TITLE
fix(gpws): "too low terrain" warning on 2nd departure

### LINKED ISSUE #2235: [BUG] Terrain Warning on T/O of 2nd Leg
**Mod Version**
Master

**Describe the bug**
I’m not sure if this is a bug or if I’m doing something incorrectly. I start cold and dark and input a flight plan manually and complete it. I shut off engines at the gate and start replacing the init page info for my second leg. Everything looks good, I can put a new flight plan in, it goes back to the takeoff performance phase, and takeoff inhibit displays as per normal during roll. The problem occurs when I leave the ground. I get “terrain pull up” warnings for about 1500’, like takeoff inhibit isn’t actually inhibiting. I have never tried a 3rd leg to test and this has never happened on a 1st leg. 

**To Reproduce**
1. Happens on any second leg, in the same session, on takeoff once my first flight plan competes and I’m shown the data page.

**Expected behavior**
“Terrain pull up” warning to be inhibited 

**Actual behavior**
“Terrain pull up” warning sounds from the beginning of climb out till about 1500’

**References**

**Additional context**

Was this working before/when did the issue start occurring?
 It’s been happening for a while. I’m not sure it’s ever not done it to me on that second leg.

Is this a problem in the vanilla unmodded game? 
Vanilla aircraft isn’t capable enough to try and reproduce.

### PR DESCRIPTION
**Pull request by vkrizan**
Fixes #2235

## Summary of Changes

`Mode4MaxRAAlt` wasn't reset after touchdown leading to "loo low terrain" warning after second departure.

## Additional context

Max RA Alt

Discord username (if different from GitHub): coolll.sk#0022

## Testing instructions

* after landing try departing again -> the "too low terrain" call out should not be heard (note: this change)
* try getting close to ground (airborne with GPWS sys on and flaps retracted) 
 * "too low terrain" should work nominally between 10 - 2450 AGL
 * try climbing out of it and check if the warning stops
 * try approach with incorrect flaps settings, warning should still be heard

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by vkrizan** [comment]
Hi @lukecologne, this is now up for review/test. Thanks.

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js [hunk 1]
```diff
@@ -88,6 +88,7 @@ class A32NX_GPWS {
             this.Mode5Code = 0;
 
             this.Mode3MaxBaroAlt = NaN;
+            this.Mode4MaxRAAlt = NaN;
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js:91]: The max RA alt gained is not supposed to be deleted if the aircraft is above the threshold (it should be saved until landing). If the aircraft is below the threshold though (i.e. on the ground), resetting it is o.k, so I would just add a check if RA is below 10 ft.
See here as a reference from the FCOM:
grafik

**Review by lukecologne — CHANGES_REQUESTED** [review]

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js (hunk 1 above)
Comment (vkrizan) [reply A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js]: Thanks for the reference.

I've updated it to reset only on RA lower than 10 or `onGround`. I'm not sure how reliable the `SIM ON GROUND`, hence the additional RA comparison.

**Review by lukecologne — APPROVED** [review]

**Comment by marcman86** [comment]
Name: marcman86#4907
Date of testing: 19.2.2021
Version of the sim: 1.13.16.0
PR Tested: #3574
Tier of testing: 1
Changes to observe: No terrain callouts on second departure
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned APP at EDDP
2.) Land
3.) Set start config (Flaps 1 and so on)
4.) Start again: no "too low terrain" 
5.) Below 1000ft AGL: "too low terrain" 
6.) Above 1000ft AGL: "too low terrain" stops
7.) Land with Flaps 1 about 800ft AGL: "too low terrain" 

Issues: None

Overall Rating: Good!

Conclusions of the testing: Seems to work

Hope this helps :-)

**Comment by Samueleonard** [comment]
Category: QA Trial
Name: 1samueleonard1#0001
Date of testing: Current date DD/MM/2021
Version of the sim: 1.13.16.0
PR Tested: #3574 
Tier of testing: 1
Changes to observe:
.verify that there is no "loo low terrain" warning after second departure.

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. loaded at PMDY, default livery, live time/weather
2. took off as normal, did a circuit and landed
3. turned around on runway, took off again,
4. did NOT get 'too low, gear' - as expected

Flight notes (When done a full flight):
works as intended

Media: Insert screenshots, recordings, or anything relevant to the review here.
N/A
Issues (If Any): Write any issues
N/A
Steps to achieve the issue
N/A
Overall Rating: Good

Conclusions of the testing: Make a small descriptive to areas where it needs to be improved and recommendations.
LGTM

**Comment by sidnov** [comment]
Category: QA Tester
Name: sidnov#8337
Date of testing: 21.2.2021
Version of the sim: 1.13.16.0
PR Tested: #3574
Tier of testing: 1

### Changes to observe: Too low terrain warning fix

### Testing technique: Tryout

### Steps you've made to test the Pull Request:

1. Spawned C/D at VABB
2. Prepared aircraft for VFR circuits
3. Departed normally out of RWY27 on runway heading, climbed to 4000ft
4. Completed a circuit, landed and taxied back for departure.
5. Departed RWY27 again, no warning/callout was observed.
6. Tried incorrect flap configuration on 2nd approach, warning was heard as expected.

### Overall Rating: Good

### Conclusions of the testing: The warning/callouts come up as expected and not in case of 2nd departures. May need additional testing just to verify.

**Comment by MisterChocker** [comment]
Wasn't there an issue with the GWPS not being reset because the FMGC doesn't goes into DONE phase?
With this PR #3658, the FMGC does!

**Comment by vkrizan** [comment]
@MisterChocker, I'm not sure if the refactoring changes from #3658 would fix it. The problem seems to be that the MaxRA is only accumulated between <10, 2450> AGL. Once touch down (or closer to ground) the value stays the same, and is later used once departed again.

See the update method `gpws()` and its conditions. Outside of the range mentioned or when the GPWS SYS is off the maxRA isn't updated at all. It could potentially call the `update_maxRA` even outside of the range or SYS off, but I'm not sure how reliable is the ON GROUND simvar.

**Comment by vkrizan** [comment]
@lukecologne @beheh Is this good for a merge? Thanks!

**Comment by beheh** [comment]
The change seems good from my side and if QA is happy I think it can be merged.

However, I do not know enough about GPWS to understand the full behavior here, so I'd prefer to have @lukecologne make the final merge decision.

**Pull request merged**